ssmconfig.go

```diff
@@ -129,6 +142,31 @@ func (p *Provider) getParameters(spec structSpec) (params map[string]string, inv
 	return params, invalidParams, nil
 }
 
+// Checks whether the value implements the TextUnmarshaler interface.
+func isTextUnmarshaler(f reflect.StructField) bool {
+	return reflect.PtrTo(f.Type).Implements(textUnmarshalerType)
+}
+
+// Create a new instance of the field's type and call its UnmarshalText([]byte) method.
+// Set the value after execution and fail if the method returns an error.
+func unmarshalText(f reflect.StructField, v reflect.Value, s string) error {
```

Comment:
This can be simplified quite a bit. The `Interface()` method on `reflect.Value` allows you to access the value as a regular go interface.

We also need to handle the case where the value is already a pointer to the implementation of text unmarshaled. I would handle that by accepting a `shouldAddr` argument.

My implementation of this would probably look something like:

```go
func unmarshalText(t reflect.Type, v reflect.Value, s string, shouldAddr bool) error {
	if shouldAddr {
		v = v.Addr()
	}
	if t.Kind() == reflect.Ptr && v.IsNil() {
		v.Set(reflect.New(t.Elem()))
	}
	return v.Interface().(encoding.TextUnmarshaler).UnmarshalText([]byte(s))
}
```

Reply:
I will definitely change the much simpler method invocation, thanks!

I didn't add support for pointers on purpose as all the other types don't support pointers yet. I thought this could be addressed in a new issue and possibly implemented using a more dynamic / recursive approach for all supported types.

What do you think? Should I add pointer support already or not?

Reply:
Hmmmmmm, Good point.

I think I would go ahead and add pointer support for `TextUnmarshaler`. It starts to get confusing what is and isn't supported when you don't allow pointers for interface types.

Reply:
✔️ 

I also added tests to empty non-required parameters and added another example to the readme.